### PR TITLE
11 ticket

### PR DESCRIPTION
**Pull request by Hassen-Ahmed**

### CODE REVIEW

File: models/articles.model.js [hunk 1]
```diff
@@ -9,25 +8,64 @@ exports.selectArticlesById = (article_id) => {
...
-    FROM articles 
-    LEFT JOIN comments ON comments.article_id = articles.article_id 
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC
-    ;
-    `;
-    return db.query(query).then(({ rows }) => {
-        return rows;
-    });
+exports.selectAllArticles = async (topic, sort_by = "created_at", order) => {
```
Comment (johnhutcheon) [inline models/articles.model.js]: order should also have a default value here 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -353,4 +353,128 @@ describe("All requests", () => {
...
+
+        test("404 will returned and when passed valid sort_id query, but no resource found", () => {
+            return request(app)
+                .get("/api/articles?sort_by=body")
+                .expect(404)
+                .then(({ body }) => {
+                    const { msg } = body;
+                    expect(msg).toBe("Not found!");
+                });
+        });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:478]: Really nice, thorough testing. Well done! 💯

**Review by johnhutcheon — APPROVED** [review]
Great work Hassen! 🙌 Happy for you to merge, just make sure you set a default value on the order as well 👍

**Comment by Hassen-Ahmed** [comment]
Thank you so much. You and you co have giving me Great and warm feedback!

**Pull request merged**